src/output_window.rs

```diff
@@ -109,7 +109,7 @@ impl OutputWindow {
             }
         } else {
             // only one copy is needed if there is no wrap around.
-            copied = input.copy_to(&mut self.window[self.end..][..tail_len]);
+            copied = input.copy_to(&mut self.window[self.end..][..length]);
```

Comment:
Could use one slice instead of two:

```suggestion
            copied = input.copy_to(&mut self.window[self.end..length]);
```

Reply:
We should not. `[x..y]` means since x-th (inclusive) until y-th (exclusive), `[x..][..y]` means since x-th (inclusive) length y

```
fn main() {
    println!("{:?}", &[0,1,2,3,4,5][1..3]); // [1, 2]
    println!("{:?}", &[0,1,2,3,4,5][1..][..3]); // [1, 2, 3]
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a111cbb5af9e6bf3f826b280a2f3e4b8

Reply:
Thanks for explanation!

I forgot that you would need `[end..(end + length)]`